microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -152,6 +152,92 @@ def verify_dpdk_nff_go(
         # run the nff-go tests
         nff_go.run_test()
 
+    @TestCaseMetadata(
+        description="""
+           Build and run DPDK multiprocess client/server sample application.
+           Requires 3 nics since client/server needs two ports + 1 nic for LISA
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            min_nic_count=3,
+            network_interface=Sriov(),
+        ),
+    )
+    def verify_dpdk_multiprocess(
+        self, node: Node, log: Logger, variables: Dict[str, Any]
+    ) -> None:
+
+        kill = node.tools[Kill]
+        pmd = "failsafe"
+        server_app_name = "dpdk-mp_server"
+        client_app_name = "dpdk-mp_client"
+        # initialize DPDK with sample applications selected for build
+        test_kit = initialize_node_resources(
+            node,
+            log,
+            variables,
+            pmd,
+            sample_apps=[
+                "multi_process/client_server_mp/mp_server",
+                "multi_process/client_server_mp/mp_client",
+            ],
+        )
+
+        # enable hugepages needed for dpdk EAL
+        _init_hugepages(node)
+
+        # setup and run mp_server application
+        examples_path = test_kit.testpmd.dpdk_build_path.joinpath("examples")
+        server_app_path = examples_path.joinpath(server_app_name)
+        client_app_path = examples_path.joinpath(client_app_name)
+
+        # EAL -l: start server on cores 1-2,
+        # EAL -n: use 4 memory channels
+        # APP: -p : set port bitmask to port 0 and 1
+        # APP: -n : allow one client to connect
+        server_proc = node.execute_async(
+            (
+                f"{server_app_path} -l 1-2 -n 4 "
+                f"-b {node.nics.get_nic_by_index(0).pci_slot} -- -p 3 -n 1"
+            ),
+            sudo=True,
+            shell=True,
+        )
+        time.sleep(1)  # give a moment for server to start
```

Comment:
Is it really needed? The blind sleep makes trouble sometimes. Divyansh added a process stdout, it can help on waiting something happens.

BTW, it's better to put comments above the code, not behind.

Reply:
That should work! It will give a nicer error message if the server fails to init too.